types/coin.go

```diff
@@ -243,6 +243,18 @@ func (coins Coins) Validate() error {
 	}
 }
 
+func (coins Coins) isSorted() bool {
+	if len(coins) < 2 {
+		return true
+	}
```

Comment:
I think this extra check is not necessary.

Reply:
right, the loop condition will handle it.

Reply:
This is early return. A very good practice AFAICT